compiler-plugin/src/main/kotlin/it/unibo/collektive/frontend/checkers/NoAlignInsideLoop.kt

```diff
@@ -15,33 +19,167 @@ import org.jetbrains.kotlin.fir.expressions.FirWhileLoop
  */
 object NoAlignInsideLoop : FirFunctionCallChecker(MppCheckerKind.Common) {
 
-    private val safeOperators = listOf("alignedOn", "align", "dealign")
+    private val SAFE_OPERATORS = listOf("alignedOn", "align", "dealign")
 
-    private fun CheckerContext.isInsideALoopWithoutAlignedOn(): Boolean {
-        val loopElementIndex = containingElements.indexOfLast { it is FirWhileLoop }
-        val functionDeclarationIndex = containingElements.dropLast(1).indexOfLast {
-            (it as? FirSimpleFunction)?.receiverParameter?.isAggregate(session) == true
-        }
-        if (loopElementIndex == -1 || functionDeclarationIndex == -1) {
-            return false
-        }
-        return loopElementIndex > functionDeclarationIndex &&
-            containingElements
-                .drop(loopElementIndex)
-                .dropLast(1)
-                .filterIsInstance<FirFunctionCall>()
-                .map { it.calleeReference.name.asString() }
-                .none { it == "alignedOn" }
-    }
+    /**
+     * Methods used inside collections to iterate their elements.
+     */
+    val ITERATIVE_METHODS = setOf(
```

Comment:
I considered that option in the beginning, however I wasn't really able to get all the methods listed here because their declaration is spread across Kotlin standard library. Most of them are implemented as extension functions, sometimes of the class `Iterable` and sometimes other classes like `List`, `Collection`, `Map` etc.

If you can give some advice on how to obtain them, it would be very helpful.

Reply:
@DanySK any suggestion on this side?